src/main/java/io/jenkins/plugins/pipelinegraphview/consoleview/PipelineConsoleViewAction.java

```diff
@@ -56,12 +62,13 @@ public void getSteps(StaplerRequest req, StaplerResponse rsp) throws IOException
         String nodeId = req.getParameter("nodeId");
         if (nodeId != null) {
             LOGGER.log(Level.FINE, "getSteps was passed nodeId '" + nodeId + "'.");
-            PipelineStepApi stepApi = new PipelineStepApi(target, nodeId);
             ObjectMapper mapper = new ObjectMapper();
-            LOGGER.log(Level.FINE, "Steps: '" + mapper.writeValueAsString(stepApi.getSteps()) + "'.");
-            rsp.getWriter().append(mapper.writeValueAsString(stepApi.getSteps()));
+            PipelineStepList steps = stepApi.getSteps(nodeId);
+            LOGGER.log(Level.FINE, "Steps: '" + mapper.writeValueAsString(steps) + "'.");
```

Comment:
should be wrapped in either a supplier or an `if LOGGER.isLoggable` pattern so the object isn't converted to a string when not needed